# TFM: Parte de preprocesamiento de datos.
### Autores: 
- ####  Adrian
- ####  Carlos Mendívil Gómez
- ####  Guillermo
- ####  Jose

In [1]:
import numpy as np
import pandas as pd 
import matplotlib.pyplot as plt
import seaborn as sns

import warnings

warnings.filterwarnings("ignore")

In [2]:
url = "https://raw.githubusercontent.com/guille1006/TFM/refs/heads/main/data2/total_data.csv"
df = pd.read_csv(url)

# Descripción general de las variables

<table>
  <tr><th>Campo</th><th>Variable valida</th><th>Tipo</th><th>Descripción</th></tr>
  
  <tr><td colspan="4" align="center"><b> Dirección del inmueble</b></td></tr>
  <tr><td>latitude</td><td>Si</td><td>float</td><td>Coordenada de latitud</td></tr>
  <tr><td>longitude</td><td>Si</td><td>float</td><td>Coordenada de longitud</td></tr>
  <tr><td>showAddress</td><td>Si</td><td>bool</td><td>Muestra la dirección o no</td></tr>
  <tr><td>address</td><td>Si</td><td>category</td><td>Calle, número y país del inmueble</td></tr>
  <tr><td>neighborhood</td><td>Si</td><td>category</td><td>Vecindario donde está ubicado el inmueble</td></tr>
  <tr><td>district</td><td>Si</td><td>category</td><td>Distrito donde está ubicado el inmueble</td></tr>
  <tr><td>municipality</td><td>Si</td><td>category</td><td>Municipio donde está ubicado el inmueble</td></tr>
  <tr><td>province</td><td>No</td><td>category</td><td>Provincia donde está ubicado el inmueble</td></tr>
  <tr><td>country</td><td>No</td><td>category</td><td>País donde está ubicado el inmueble</td></tr>

  <tr><td colspan="4" align="center"><b> Características propias del inmueble</b></td></tr>
  <tr><td>rooms</td><td>Si</td><td>int</td><td>Número de habitaciones del inmueble</td></tr>
  <tr><td>size</td><td>Si</td><td>float</td><td>Metros cuadrados del inmueble</td></tr>
  <tr><td>bathrooms</td><td>Si</td><td>int</td><td>Número de baños</td></tr>
  <tr><td>exterior</td><td>Si</td><td>bool</td><td>Tiene parte exterior o no</td></tr>
  <tr><td>status</td><td>Si</td><td>category</td><td>Estado de la vivienda</td></tr>
  <tr><td>detailedType.subTypology</td><td>Si</td><td>category</td><td>Subtipo de inmueble</td></tr>
  <tr><td>detailedType.typology</td><td>Si</td><td>category</td><td>Tipo de inmueble</td></tr>
  <tr><td>floor</td><td>Si</td><td>int</td><td>Altura del piso</td></tr>
  <tr><td>hasLift</td><td>Si</td><td>bool</td><td>Tiene ascensor o no</td></tr>
  <tr><td>parkingSpace.hasParkingSpace</td><td>Si</td><td>bool</td><td>Tiene espacio para aparcar o no</td></tr>
  <tr><td>parkingSpace.isParkingSpaceIncludedInPrice</td><td>Si</td><td>bool</td><td>Está el precio del aparcamiento incluido en el precio o no</td></tr>
  <tr><td>parkingSpace.parkingSpacePrice</td><td>Si</td><td>float</td><td>Precio del espacio de aparcamiento</td></tr>
  <tr><td>price</td><td>Si</td><td>float</td><td>Precio del inmueble</td></tr>
  <tr><td>priceInfo.price.amount</td><td>No</td><td>float</td><td>Precio del inmueble</td></tr>
  <tr><td>priceByArea</td><td>Si</td><td>float</td><td>Precio del metro cuadrado del inmueble</td></tr>
  <tr><td>priceInfo.price.currencySuffix</td><td>No</td><td>category</td><td>Moneda del precio</td></tr>
  <tr><td>priceInfo.price.priceDropInfo.formerPrice</td><td>Si</td><td>float</td><td>Precio anterior del inmueble</td></tr>
  <tr><td>priceInfo.price.priceDropInfo.priceDropPercentage</td><td>Si</td><td>float</td><td>Porcentaje de rebaja del precio con respecto al precio anterior</td></tr>
  <tr><td>priceInfo.price.priceDropInfo.priceDropValue</td><td>Si</td><td>float</td><td>Cantidad del dinero que se ha rebajado con respecto al precio anterior</td></tr>
  <tr><td>newDevelopment</td><td>Si</td><td>bool</td><td>Es vivienda de nueva construcción o no</td></tr>
  <tr><td>newDevelopmentFinished</td><td>Si</td><td>bool</td><td>Es vivienda de nueva construcción terminada o no</td></tr>
  <tr><td>topNewDevelopment</td><td>No</td><td>bool</td><td>Es un inmueble de nueva construcción o no</td></tr>
  <tr><td>highlight.groupDescription</td><td>Si</td><td>category</td><td></td></tr>

  <tr><td colspan="4" align="center"><b> Información relacionada con el anuncio</b></td></tr>
  <tr><td>description</td><td>Si</td><td>category</td><td>Descripción del anuncio, escrita por el dueño o la inmobiliaria</td></tr>
  <tr><td>numPhotos</td><td>Si</td><td>int</td><td>Número de fotos</td></tr>
  <tr><td>has360</td><td>Si</td><td>bool</td><td>Tiene imagen 360º del piso o no</td></tr>
  <tr><td>has3DTour</td><td>Si</td><td>bool</td><td>Tiene tour 3D o no</td></tr>
  <tr><td>hasVideo</td><td>Si</td><td>bool</td><td>Tiene video o no</td></tr>
  <tr><td>hasPlan</td><td>Si</td><td>bool</td><td>Tiene vista en plano o no</td></tr>
  <tr><td>hasStaging</td><td>Si</td><td>bool</td><td>Tiene visualización HomeStaging o no</td></tr>
  <tr><td>operation</td><td>No</td><td>category</td><td>Tipo de operación: Venta, alquiler...</td></tr>
  <tr><td>suggestedTexts.subtitle</td><td>Si</td><td>object</td><td>Subtítulo del anuncio</td></tr>
  <tr><td>suggestedTexts.title</td><td>Si</td><td>object</td><td>Título del anuncio</td></tr>

  <tr><td colspan="4" align="center"><b> Información propia de Idealista</b></td></tr>
  <tr><td>externalReference</td><td>No</td><td>object</td><td>Referencia externa usada por la página web de Idealista</td></tr>
  <tr><td>thumbnail</td><td>No</td><td>object</td><td>Enlace a una imagen del anuncio</td></tr>
  <tr><td>topPlus</td><td>Si</td><td>bool</td><td>Pertenece al plan Plus o no</td></tr>
  <tr><td>highlight.groupDescription</td><td>Si</td><td>category</td><td>Tipo de grupo de descripcion</td></tr>
  <tr><td>url</td><td>No</td><td>object</td><td>URL del anuncio</td></tr>
  <tr><td>notes</td><td>No</td><td>list</td><td>Lista vacía, no sabemos su uso</td></tr>
</table>


### Explicación de las variables no usadas:
- **Identificadores y Metadatos:** Variables como `url`, `externalReference` y `thumbnail`, que actúan como identificadores únicos o enlaces y no contienen información generalizable.

- **Variables Constantes:** Campos con un valor único para toda la muestra, como `country` (Spain), `province` (Madrid), `operation` (sale) y priceInfo.price.currencySuffix (EUR), que carecen de poder discriminatorio.

- **Variables Redundantes:** El campo `priceInfo.price.amount`, que es un duplicado exacto de la variable `price`.

- **Variables sin Varianza:** Columnas como `notes` y `TopNewDevelopment`, que se encontraron vacías, nulas o con un valor constante (False) en la totalidad de los registros.

# Limpieza de datos
El proceso de **limpieza de datos** se inicia con una fase de **análisis exploratorio preliminar**, cuyo objetivo principal es obtener una visión global del conjunto de datos almacenado en el *DataFrame*.  

In [3]:
# Vamos a mostrar una visión general del dataframe, para ver la distribución de las variables
print(df.shape)
df.describe().T

(26267, 49)


,count,mean,std,min,25%,50%,75%,max
bathrooms,26267.0,2.346442e+00,1.557807e+00,0.000000,1.000000e+00,2.000000e+00,3.000000e+00,2.900000e+01
latitude,26267.0,4.043244e+01,1.148277e-01,40.001899,4.038818e+01,4.042643e+01,4.046575e+01,4.110693e+01
longitude,26267.0,-3.717539e+00,1.638774e-01,-4.494274,-3.769397e+00,-3.698660e+00,-3.656073e+00,-3.071336e+00
numPhotos,26267.0,2.919210e+01,1.947732e+01,0.000000,1.600000e+01,2.600000e+01,3.900000e+01,1.900000e+02
parkingSpace.parkingSpacePrice,414.0,3.064735e+04,2.041348e+04,60.000000,1.500000e+04,2.800000e+04,4.000000e+04,1.000000e+05
price,26267.0,8.877257e+05,1.177656e+06,11000.000000,2.699450e+05,4.890000e+05,1.050000e+06,1.850000e+07
priceByArea,26267.0,5.173715e+03,3.526807e+03,2.000000,2.650000e+03,3.990000e+03,6.860000e+03,3.714300e+04
priceInfo.price.amount,26267.0,8.877257e+05,1.177656e+06,11000.000000,2.699450e+05,4.890000e+05,1.050000e+06,1.850000e+07
priceInfo.price.priceDropInfo.formerPrice,2806.0,6.127779e+05,5.973913e+05,32000.000000,2.500000e+05,3.990000e+05,7.500000e+05,7.500000e+06
priceInfo.price.priceDropInfo.priceDropPercentage,2806.0,4.614754e+00,3.116659e+00,1.000000,3.000000e+00,4.000000e+00,6.000000e+00,4.500000e+01


In [4]:
# Voy a realizar un estudio de las variables y del tipo de estas
info_types = pd.DataFrame(
    {"Tipo de columna": df.dtypes, 
     "Valores únicos": df.nunique(dropna=False),
     "Valores nulos": df.isnull().sum()}
).sort_values(by="Valores nulos", ascending=False)

info_types

,Tipo de columna,Valores únicos,Valores nulos
parkingSpace.parkingSpacePrice,float64,67,25853
newDevelopmentFinished,object,3,23536
priceInfo.price.priceDropInfo.formerPrice,float64,830,23461
priceInfo.price.priceDropInfo.priceDropValue,float64,276,23461
priceInfo.price.priceDropInfo.priceDropPercentage,float64,27,23461
detailedType.subTypology,object,14,17080
highlight.groupDescription,object,4,14883
parkingSpace.isParkingSpaceIncludedInPrice,object,3,14374
parkingSpace.hasParkingSpace,object,2,14374
neighborhood,object,198,10291


El siguiente paso en el proceso de depuración consistió en **descartar aquellas variables previamente identificadas como prescindibles**. 

De este modo, se garantiza que el *DataFrame* resultante contenga únicamente las variables útiles y pertinentes para los objetivos del estudio.  


In [ ]:
# Columnas que no son necesarias para la limpieza de datos inicial
cols_to_drop = [
    'url', 'externalReference', 'thumbnail', 'propertyCode', 
    'description', 'suggestedTexts.title', 'suggestedTexts.subtitle',
    'notes', 'address', 'country', 'province', 'operation', 
    'priceInfo.price.currencySuffix', 'priceInfo.price.amount',
    'priceInfo.price.priceDropInfo.priceDropValue', 'priceInfo.price.priceDropInfo.formerPrice',
    'topNewDevelopment', 'latitude', 'longitude'
]

df.drop(columns=cols_to_drop, inplace=True)

column_tot = df.columns

Otro aspecto fundamental dentro del proceso de limpieza de datos consiste en **identificar los posibles valores que puede adoptar cada variable categórica y realizar un conteo de su frecuencia de aparición**. Para ello usaremos la función de `tabla_valres_unicos`

In [ ]:
def tabla_valores_unicos(df):
    """
    Devuelve un DataFrame con la frecuencia absoluta y relativa de los valores únicos 
    en cada columna de `df`.

    Los valores nulos se muestran como "Nulo". El resultado tiene un MultiIndex 
    (columna, valor) y columnas:
        - "n": frecuencia absoluta
        - "%": frecuencia relativa (0 a 1)

    Args:
        df (pandas.DataFrame): DataFrame de entrada.

    Returns:
        pandas.DataFrame: Tabla de frecuencias por columna y valor único.
    """
    df_frec = pd.DataFrame()

    for col in df.columns:
        # Generare, solo para la visualización, que estos elementos se llamen "Nulo"
        series_fillna = df[col].fillna("Nulo")

        # Realizamos el conteo de cada valor y su frecuencia
        series_cont = series_fillna.value_counts()
        series_frec = series_cont/series_cont.sum()

        # Usaremos un DataFrame con MultiIndex para poder visualizar y acceder mejor a cada columna
        arrays_multiindex = [[col]* len(series_cont), series_cont.index.tolist()]
        multi_index = pd.MultiIndex.from_arrays(arrays_multiindex)

        result = pd.DataFrame({"n":series_cont.values, "%": series_frec.values}, index=multi_index)
        df_frec = pd.concat([df_frec, result], axis=0)
    
    return df_frec

In [ ]:
df_frec = tabla_valores_unicos(df)

## Tratamiento de los valores nulos


### Variables Booleanas nulas

| Variable                                 | Nº de Nulos | Tipo   | Estrategia de Imputación                   |
| :--------------------------------------- | :---------- | :----- | :----------------------------------------- |
| **exterior** | 7038        | object | Rellenar con `False` y convertir a binaria. |
| **hasLift** | 6716        | object | Rellenar con `False` y convertir a binaria. |
| **newDevelopmentFinished** | 23536       | object | Son aquellos que son de nueva obra, podemos poner el resto como `False` |
| **parkingSpace.hasParkingSpace** | 14374       | object | Solo tiene valores `True` y `NaN`, por lo que estos últimos serán `False` |
| **parkingSpace.isParkingSpaceIncludedInPrice**| 14374    | object | Nulos iguales a `hasParkingSpace`, repetir proceso. |

In [ ]:
# exterior
df["exterior"] = df["exterior"].fillna(False)

# hasLift
df["hasLift"] = df["hasLift"].fillna(False)

# newDevelopmentFinished	
df["newDevelopmentFinished"] = df["newDevelopmentFinished"].fillna(False)

# parkingSpace.hasparkingSpace
df["parkingSpace.hasParkingSpace"] = df["parkingSpace.hasParkingSpace"].fillna(False)

# parkingSpace.isParkingSpaceIncludedInPrice
df["parkingSpace.isParkingSpaceIncludedInPrice"] = df["parkingSpace.isParkingSpaceIncludedInPrice"].fillna(False)

### Variables catégoricas nulas

| Variable | Nº de Nulos | Tipo | Estrategia de Imputación |
| :--- | :--- | :--- | :--- |
| **floor** | 7207 | object | Se estandarizan los valores no numéricos (ej. 'bj', 'st') a sus equivalentes numéricos (0, -1).<br> Los nulos se tratan como planta baja (0) y se crea una variable `floor_cat` para capturar el efecto no lineal de estas categorías especiales. |
| **detailedType.subTypology** | 17080 | object | Se imputa con una nueva categoría, 'Vacío', dada que no existe subtipolgia en ese caso. |
| **district** | 3450 | object | Se imputan los nulos mediante geolocalización inversa, utilizando las coordenadas para asignar el distrito correspondiente. |
| **highlight.groupDescription** | 14883 | object | Se imputa con la categoría 'Vacio', asumiendo que la ausencia de valor implica la ausencia de una promoción. |
| **neighborhood** | 10291 | object | La columna se elimina debido a su alta cardinalidad y nulos, y se reemplaza por una feature binaria, `has_neighborhood`. |
| **status** | 597 | object | Se imputan los nulos como 'New development', basándose en la correlación observada con la variable `newDevelopment`, lo que resuelve una inconsistencia en los datos. |

In [ ]:
# floor
def floor_function(value):
    numeric_cat = "Numerica"
    x = value["floor"]

    len_x = len(x)

    numeric = [c for c in x if c.isdigit()]
    alnum = [c for c in x if not c.isalnum()]

    if len(numeric) == len_x:
        res_num = float(x)
        res_cat = numeric_cat

    elif alnum:
        if alnum[0] in [".", ","]:
            res_num = float(x)
        elif alnum[0] == "+":
            res_num = float("".join(numeric))
        elif alnum[0] == "-":
            res_num = float(x)
        res_cat = numeric_cat

    else:
        res_num = 0.0
        if value["detailedType.typology"] in ["chalet", "countryHouse"]:
            res_cat = numeric_cat
        else:
            res_cat = x

    if res_num > 15:
        res_num = 15
    return pd.Series([res_num, res_cat])

In [3]:
# floor
def floor_function(value):
    numeric_cat = "Numerica"
    x = value["floor"]

    len_x = len(x)

    numeric = [c for c in x if c.isdigit()]
    alnum = [c for c in x if not c.isalnum()]

    if len(numeric) == len_x:
        res_num = float(x)
        res_cat = numeric_cat

    elif alnum:
        if alnum[0] in [".", ","]:
            res_num = float(x)
        elif alnum[0] == "+":
            res_num = float("".join(numeric))
        elif alnum[0] == "-":
            res_num = float(x)
        res_cat = numeric_cat

    else:
        res_num = 0.0
        if value["detailedType.typology"] in ["chalet", "countryHouse"]:
            res_cat = numeric_cat
        else:
            res_cat = x

    if res_num > 15:
        res_num = 15
    return pd.Series([res_num, res_cat])

df["floor"] = df["floor"].fillna("Nulo")

df_floor = df.apply(floor_function, axis=1)
df["floor"], df["floor_cat"] = df_floor[0], df_floor[1]

# priceInfo.price.priceDropInfo.priceDropPercentage
df["priceInfo.price.priceDropInfo.priceDropPercentage"] = df["priceInfo.price.priceDropInfo.priceDropPercentage"].fillna(0.0)

# parkingSpace.parkingSpacePrice
df["parkingSpace.parkingSpacePrice"] = df["parkingSpace.parkingSpacePrice"].fillna(0.0)

# parkingSpace.hasparkingSpace
df["parkingSpace.hasParkingSpace"] = df["parkingSpace.hasParkingSpace"].fillna(False)

# parkingSpace.isParkingSpaceIncludedInPrice
df["parkingSpace.isParkingSpaceIncludedInPrice"] = df["parkingSpace.isParkingSpaceIncludedInPrice"].fillna(False)

# newDevelopmentFinished	
df["newDevelopmentFinished"] = df["newDevelopmentFinished"].fillna(False)

# highlight.groupDescription
df["highlight.groupDescription"] = df["highlight.groupDescription"].fillna("Vacío")

# exterior
df["exterior"] = df["exterior"].fillna(False)

# hasLift
df["hasLift"] = df["hasLift"].fillna(False)

# status
df["status"] = df["status"].fillna("newdevelopment")

# detailedType.subTypology
df["detailedType.subTypology"] = df["detailedType.subTypology"].fillna("Vacío")

In [4]:
# Columnas que no son necesarias para la limpieza de datos inicial
cols_to_drop = [
    'url', 'externalReference', 'thumbnail', 'propertyCode', 
    'description', 'suggestedTexts.title', 'suggestedTexts.subtitle',
    'notes', 'address', 'country', 'province', 'operation', 
    'priceInfo.price.currencySuffix', 'priceInfo.price.amount',
    'priceInfo.price.priceDropInfo.priceDropValue', 'priceInfo.price.priceDropInfo.formerPrice',
    'topNewDevelopment', 'latitude', 'longitude'
]

df.drop(columns=cols_to_drop, inplace=True)

column_tot = df.columns

# Voy a realizar un estudio de las variables y del tipo de estas
info_types = pd.DataFrame(
    {"Tipo de columna": df.dtypes, 
     "Valores únicos": df.nunique(dropna=False),
     "Valores nulos": df.isnull().sum()}
).sort_values(by="Valores nulos", ascending=False)

info_types

,Tipo de columna,Valores únicos,Valores nulos
neighborhood,object,198,10291
district,object,271,3450
bathrooms,int64,19,0
numPhotos,int64,144,0
topPlus,bool,2,0
status,object,3,0
size,float64,927,0
showAddress,bool,2,0
rooms,int64,22,0
propertyType,object,6,0


In [5]:
def tabla_valores_unicos(df):
    df_frec = pd.DataFrame()

    for col in df.columns:
        # generare, solo para la visualización que estos elementos se llamen "Nulo"
        series_fillna = df[col].fillna("Nulo")

        # 
        series_cont = series_fillna.value_counts()
        series_frec = series_cont/series_cont.sum()

        #
        arrays_multiindex = [[col]* len(series_cont), series_cont.index.tolist()]
        multi_index = pd.MultiIndex.from_arrays(arrays_multiindex)

        result = pd.DataFrame({"n":series_cont.values, "%": series_frec.values}, index=multi_index)
        df_frec = pd.concat([df_frec, result], axis=0)
    
    return df_frec

In [6]:
df_frec = tabla_valores_unicos(df)

# Diferenciacion de variables

In [7]:
from itertools import zip_longest
# variables booleanas
variable_bool = info_types[info_types["Valores únicos"]<3].index.to_list()

# variables numericas
variables_num = df.select_dtypes(include=["number"]).columns.tolist()

# variables categoricas
categoricas = []
for col in df.columns:
    if df[col].dtypes == "object":
        if df[col].nunique(dropna=False) <20:
            categoricas.append(col)
df[categoricas] = df[categoricas].astype("category")
variables_cat = df.select_dtypes(include=["category"]).columns.tolist()


print(f"{"variable_bool":50}{"variables_num":<50}{"variables_cat":<10}")
print("------------------------------------------------------------------------------------------------------------------------")
for variable in zip_longest(variable_bool, variables_num, variables_cat, fillvalue=""): 
    print(f"{variable[0]:50}{variable[1]:<45}{variable[2]:<10}")

variable_bool                                     variables_num                                     variables_cat
------------------------------------------------------------------------------------------------------------------------
topPlus                                           bathrooms                                    detailedType.subTypology
showAddress                                       floor                                        detailedType.typology
parkingSpace.isParkingSpaceIncludedInPrice        numPhotos                                    highlight.groupDescription
parkingSpace.hasParkingSpace                      parkingSpace.parkingSpacePrice               propertyType
newDevelopment                                    price                                        status    
newDevelopmentFinished                            priceByArea                                  floor_cat 
hasVideo                                          priceInfo.price.priceDropInfo.priceD

In [8]:
sum_var = 0
variables = variable_bool + variables_num + variables_cat


print(f"Columnas totales: {len(column_tot)}")
print(f"Variables: {len(variables)}")
print("Variables no usadas:")
print([x for x in column_tot if x not in variables])

Columnas totales: 31
Variables: 28
Variables no usadas:
['district', 'municipality', 'neighborhood']


In [9]:
"""
for var in variables_num:
    show = df_copy.sort_values(by=var)
    print(var)
    print(show[var].head(25))
"""

'\nfor var in variables_num:\n    show = df_copy.sort_values(by=var)\n    print(var)\n    print(show[var].head(25))\n'

In [10]:
"""  
def graph_hist(series, bins="auto"):
    plt.hist(series, bins=bins)
    plt.title(series.name)
    plt.show()

for var in variables_num:
    graph_hist(df_copy[var])
"""

'  \ndef graph_hist(series, bins="auto"):\n    plt.hist(series, bins=bins)\n    plt.title(series.name)\n    plt.show()\n\nfor var in variables_num:\n    graph_hist(df_copy[var])\n'

# Onehot encoder de las categoricas

In [11]:
df_encoded = pd.get_dummies(df, columns=variables_cat, drop_first=True)

In [12]:
for col in df_encoded.columns:
    print(col)

bathrooms
district
exterior
floor
has360
has3DTour
hasLift
hasPlan
hasStaging
hasVideo
municipality
neighborhood
newDevelopment
newDevelopmentFinished
numPhotos
parkingSpace.hasParkingSpace
parkingSpace.isParkingSpaceIncludedInPrice
parkingSpace.parkingSpacePrice
price
priceByArea
priceInfo.price.priceDropInfo.priceDropPercentage
rooms
showAddress
size
topPlus
detailedType.subTypology_casaDePueblo
detailedType.subTypology_casale
detailedType.subTypology_caseron
detailedType.subTypology_cortijo
detailedType.subTypology_countryHouse
detailedType.subTypology_duplex
detailedType.subTypology_independantHouse
detailedType.subTypology_masia
detailedType.subTypology_palacio
detailedType.subTypology_penthouse
detailedType.subTypology_semidetachedHouse
detailedType.subTypology_studio
detailedType.subTypology_terracedHouse
detailedType.typology_countryHouse
detailedType.typology_flat
highlight.groupDescription_Top
highlight.groupDescription_Top+
highlight.groupDescription_Vacío
propertyType_count

In [13]:
len(df_encoded.columns)

56

In [14]:
df_encoded.to_csv("data2/data_fin.csv", index=False)

In [15]:
df.to_csv("data2/data_complete.csv", index=False)